### PR TITLE
Add Person, Student and Teacher classes to the School Library App

### PR DESCRIPTION
**Pull request by JonahKayizzi**
This pull request consists of the following;

- Created classes Person, Student and Teacher with their respective variables and methods

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def of_age?
+    @age >= 18
+  end
```
Comment (Nemwel-Boniface) [inline person.rb:15]: - [ ] Kindly make this method a private method as it is in the project requirements.

image

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @JonahKayizzi      :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project so far! :clap:
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (JonahKayizzi) [reply person.rb:15]: Thanks for this. It has been corrected.

**Review by ichala — APPROVED** [review]
Hi @JonahKayizzi   ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**